cmd/lndinit/cmd_init_wallet.go

```diff
@@ -0,0 +1,292 @@
+		seed, err = readFile(x.File.Seed)
+		if err != nil {
+			return err
+		}
+
+		// The seed passphrase is optional.
+		if x.File.SeedPassphrase != "" {
+			log("Reading seed passphrase from file")
+			seedPassPhrase, err = readFile(x.File.SeedPassphrase)
+			if err != nil {
+				return err
+			}
+		}
+
+		log("Reading wallet password from file")
+		walletPassword, err = readFile(x.File.WalletPassword)
+		if err != nil {
+			return err
+		}
+
+	// Read passphrase from Kubernetes secret.
+	case storageK8s:
+		k8sSecret := &k8sSecretOptions{
+			Namespace:  x.K8s.Namespace,
+			SecretName: x.K8s.SecretName,
+		}
+		k8sSecret.SecretEntryName = x.K8s.SeedEntryName
+
+		log("Reading seed from k8s secret %s (namespace %s)",
+			x.K8s.SecretName, x.K8s.Namespace)
+		seed, _, err = readK8s(k8sSecret)
+		if err != nil {
+			return err
+		}
+
+		// The seed passphrase is optional.
+		if x.K8s.SeedPassphraseEntryName != "" {
+			log("Reading seed passphrase from k8s secret %s "+
+				"(namespace %s)", x.K8s.SecretName,
+				x.K8s.Namespace)
+			k8sSecret.SecretEntryName = x.K8s.SeedPassphraseEntryName
+			seedPassPhrase, _, err = readK8s(k8sSecret)
+			if err != nil {
+				return err
+			}
+		}
+
+		log("Reading wallet password from k8s secret %s (namespace %s)",
+			x.K8s.SecretName, x.K8s.Namespace)
+		k8sSecret.SecretEntryName = x.K8s.WalletPasswordEntryName
+		walletPassword, _, err = readK8s(k8sSecret)
+		if err != nil {
+			return err
+		}
+
+	}
+
+	// Decrypt the seed now to make sure we got valid data before we
+	// check anything else.
+	seedWords := strings.Split(seed, " ")
+	if len(seedWords) != aezeed.NumMnemonicWords {
+		return fmt.Errorf("invalid seed, expected %d words but got %d",
+			aezeed.NumMnemonicWords, len(seedWords))
+	}
+	var seedMnemonic aezeed.Mnemonic
+	copy(seedMnemonic[:], seedWords)
+	cipherSeed, err := seedMnemonic.ToCipherSeed([]byte(seedPassPhrase))
+	if err != nil {
+		return fmt.Errorf("error decrypting seed with passphrase: %v",
+			err)
+	}
+
+	// The output directory must be specified explicitly. We don't want to
+	// assume any defaults here!
+	walletDir := lncfg.CleanAndExpandPath(x.OutputWalletDir)
+	if walletDir == "" {
+		return fmt.Errorf("must specify output wallet directory")
+	}
+	if strings.HasSuffix(walletDir, ".db") {
+		return fmt.Errorf("output wallet directory must not be a file")
+	}
+
+	// The wallet directory must either not exist yet or be a directory.
+	stat, err := os.Stat(walletDir)
+	switch {
+	case os.IsNotExist(err):
+		err = os.MkdirAll(walletDir, defaultDirPermissions)
+		if err != nil {
+			return fmt.Errorf("error creating directory %s: %v",
+				walletDir, err)
+		}
+
+	case !stat.IsDir():
+		return fmt.Errorf("output wallet directory must not be a file")
+	}
+
+	// We should now be able to properly determine if a wallet already
+	// exists or not. Depending on the flags, we either create or validate
+	// the wallet now.
+	walletFile := filepath.Join(walletDir, wallet.WalletDBName)
```

Comment:
Why not use `SubscribeState` and `InitWallet` instead to create the wallet directly in LND?

Reply:
Yeah, we need initialization over RPC for two reasons:
 - For remote databases like etcd and Postgres where the wallet isn't a file anymore
 - For initializing a watch-only wallet where we also need to pass along the accounts

So I went ahead and added the `--init-type` flag which you can set to `file` or `rpc`.
I think we still want to keep the `file` version for the most paranoid setup (because spinning up an non-authenticated RPC server and receiving a seed over it is a potential security hole).